src/TorchSharp/TorchVision/Image.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+using static TorchSharp.torch;
+
+namespace TorchSharp.torchvision
+{
+    public static partial class io
+    {
+
+        public static IImager DefaultImager { get; set; } = new ImagerSharp();
+
+        public enum ImageFormat
+        {
+            PNG,
+            JPEG,
+            UNKNOWN
+        }
+        public interface IImager
```

Comment:
>interface [](http://example.com/codeflow?start=15&length=9)

Is it expected to extend this interface in the future? If so, this can be issue for .NET Framework.

You can just make this as abstract class which give flexibility to extend in the future.

Reply:
Could you explain a bit or point to where I can learn why this would be an issue? I'll change it to an abstract class, I'm just not familiar enough with .NET to see the problem.

Reply:
[Default Interface Methods](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/default-interface-methods?ranMID=43674&ranEAID=rl2xnKiLcHs&ranSiteID=rl2xnKiLcHs-YlRlaRrWTrEFNgBuwYx6TQ&epi=rl2xnKiLcHs-YlRlaRrWTrEFNgBuwYx6TQ&irgwc=1&OCID=AID2200057_aff_7795_1243925&tduid=(ir__ixfocyqv29kfqmcc1aulhv0fbf2xv1wna3xvqwfy00)(7795)(1243925)(rl2xnKiLcHs-YlRlaRrWTrEFNgBuwYx6TQ)()&irclickid=_ixfocyqv29kfqmcc1aulhv0fbf2xv1wna3xvqwfy00#summary) is the feature added at some point for .NET Core. But this is not supported on the .NET Framework. If any user build any library/app running on .NET Framework, then the interface gets extended in the future, such library or app can break because of that.